File: HelperJobby/DTOs/Account/EmployerDTO.cs
using HelperJobby.DTOs.Job;
using HelperJobby.DTOs.Messaging;
using HelperJobby.DTOs.Organization;
using HelperJobby.DTOs.User;

namespace HelperJobby.DTOs.Account;

public class EmployerDTO
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Email { get; set; }
    public string ContactNumber { get; set; }
    public int HasPostedFirstJob { get; set; }
    public string FullName { get; set; }
    public UserDTO User { get; set; }
    public int OrganizationId { get; set; }
    public OrganizationDTO Organization { get; set; }
    public List<JobDTO> Jobs { get; set; }
    public List<IncompleteJobDTO> IncompleteJobs { get; set; }
    public List<ConversationDTO> Conversations { get; set; }
}

File: HelperJobby/client-app/src/DTOs/accountDTOs/EmployerDTO.ts
import {OrganizationDTO} from "../organizationDTOs/OrganizationDTO";
import {JobDTO} from "../jobRelatetedDTOs/JobDTO";
import {UserDTO} from "../userRelatedDTOs/UserDTO";
import {IncompleteJobDTO} from "../jobRelatetedDTOs/IncompleteJobDTO";
import {ConversationDTO} from "../MessagingDTOs/ConversationDTO";

export interface EmployerDTO {
    id: number;
    userId: number;
    email: string;
    contactNumber: string;
    hasPostedFirstJob: boolean;
    fullName: string;
    user: UserDTO;
    organizationId: number;
    organization: OrganizationDTO;
    jobs: JobDTO[];
    incompleteJobs: IncompleteJobDTO[];
    conversations: ConversationDTO[];
}